src/main/java/com/esri/samples/editing/edit_and_sync_features/EditAndSyncFeaturesSample.java

```diff
@@ -0,0 +1,403 @@
+          Point2D screenPoint = new Point2D(event.getX(), event.getY());
+
+          // get map location corresponding to screen point
+          Point mapPoint = mapView.screenToLocation(screenPoint);
+
+          // iterate through all feature layers in the map
+          for (Layer layer : mapView.getMap().getOperationalLayers()) {
+            final FeatureLayer featureLayer = (FeatureLayer) layer;
+
+            // identify any clicked feature
+            ListenableFuture<IdentifyLayerResult> results = mapView.identifyLayerAsync(featureLayer, screenPoint, 1, false, 1);
+            results.addDoneListener(() -> {
+              try {
+                // get selected feature, if it is a point
+                List<GeoElement> elementList = results.get().getElements();
+                if (elementList.size() > 0 && elementList.get(0) instanceof ArcGISFeature && elementList.get(0).getGeometry().getGeometryType() == GeometryType.POINT) {
+
+                  // clicked on a feature, select it
+                  ArcGISFeature selectedFeature = (ArcGISFeature) elementList.get(0);
+                  // TODO: this doesn't clear the selection in all FL, hence some elements stay selected?
+                  featureLayer.clearSelection(); // clear previous selection
+                  featureLayer.selectFeature(selectedFeature);
+
+                  // change the state to editing
+                  currentEditState = EditState.EDITING;
+
+                } else {
+                  // didn't click on a feature
+                  ListenableFuture<FeatureQueryResult> selectedQuery = featureLayer.getSelectedFeaturesAsync();
+                  selectedQuery.addDoneListener(() -> {
+                    try {
+                      // check if a feature is currently selected
+                      FeatureQueryResult selectedQueryResult = selectedQuery.get();
+                      Iterator<Feature> features = selectedQueryResult.iterator();
+                      if (features.hasNext()) {
+                        // move selected feature to clicked location
+                        ArcGISFeature selectedFeature = (ArcGISFeature) features.next();
+                        selectedFeature.loadAsync();
+                        selectedFeature.addDoneLoadingListener(() -> {
+                          if (selectedFeature.canUpdateGeometry()) {
+                            // apply the edits
+                            selectedFeature.setGeometry(mapPoint);
+                            selectedFeature.getFeatureTable().updateFeatureAsync(selectedFeature);
+
+                            // refresh ui to enable syncing
+                            currentEditState = EditState.READY;
+                            geodatabaseButton.setText("Sync Geodatabase");
+                            geodatabaseButton.setDisable(false);
+                          }
+                        });
+                      } // else nothing currently selected, do nothing
```

Comment:
I don't think a comment is needed here, unless it is referring to the code on line 213?